.github/workflows/main.tf

```diff
@@ -0,0 +1,35 @@
+provider "aws" {
+  region = var.aws_region
+}
+
+terraform {
+  backend "s3" {
+    bucket         = "your-s3-bucket-name"
+    key            = "terraform.tfstate"
+    region         = var.aws_region
+    encrypt        = true
+    dynamodb_table = "terraform-lock"
+  }
+}
+
+resource "aws_instance" "ec2_instance" {
+  ami           = var.ami_id
+  instance_type = var.instance_type
+  key_name      = var.key_name
+
+  tags = {
+    Name = "MyEC2Instance"
+  }
+}
+
+  user_data = <<-EOF
+    #!/bin/bash
+    ssh -i ~/my_pem.pem ubuntu@current_instance_ip
+    mv /home/ubuntu/linc-cv /home/ubuntu/linc-cv-backup
+    aws s3 cp s3://linc-backup/linc-cv/linc-cv-20181223.tar.gz /home/ubuntu/linc-cv-backup/linc-cv-20181223.tar.gz
```

Comment:
so no need to access this s3 bucket altogether?

Reply:
after doing some research we can clone the linc-cv by adding:

`git clone https://TOKEN@@github.com:linc-lion/linc-cv.git /path/to/updated/linc-cv`

we can store the token in a github secret as well or in aws secrets manager grab it like so in the user_data script:

`TOKEN=$(aws secretsmanager get-secret-value --secret-id your-secret-id --query 'SecretString' --output text)`

Reply:
> so no need to access this s3 bucket altogether?

Correct. I uploaded the latest code to the AWS instance manually from my laptop during deployment. But, before I do that, I use s3 copy to backup the existing production code. Now that we have a deployment pipeline, we should be able to correlate the deployment with the code. Before deployment, can we tag a release and then clone from that release in the user-data?

Reply:
> after doing some research we can clone the linc-cv by adding:
> 
> `git clone https://TOKEN@@github.com:linc-lion/linc-cv.git /path/to/updated/linc-cv`
> 
> we can store the token in a github secret as well or in aws secrets manager grab it like so in the user_data script:
> 
> `TOKEN=$(aws secretsmanager get-secret-value --secret-id your-secret-id --query 'SecretString' --output text)`

It is good to have options. Since the secrets are used by Github for deployment, I vote to keep the secret on Github, with encryption. @habibam @tbusath8 What do you think?

Reply:
@tbusath8 we could replace the TOKEN value with `TOKEN="${{ secrets.YOUR_SECRET_NAME }}"` if we use github secrets

Reply:
@shihgianlee since linc-cv is a public repo, I want do more research on how to protect who can trigger the action.

Reply:
I agree, keeping the altogether is a good idea